xrpl/utils/mptoken_metadata.py

```
@@ -0,0 +1,397 @@
+    if bool(HEX_REGEX.fullmatch(input_hex)) is False:
+        raise ValueError("MPTokenMetadata must be in hex format.")
+
+    try:
+        json_metadata = json.loads(hex_to_str(input_hex))
+    except Exception as e:
+        raise ValueError(
+            f"MPTokenMetadata is not properly formatted as JSON - {e}"
+        ) from e
+
+    if not isinstance(json_metadata, dict):
+        raise ValueError("MPTokenMetadata must be a JSON object.")
```

Ruff (0.14.2)

Prefer `TypeError` exception for invalid type

(TRY004)